diff --git a/office_hours.api.php b/office_hours.api.php
new file mode 100644
index 0000000..4907eb8
--- /dev/null
+++ b/office_hours.api.php
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @file
+ * Hooks and API provided by the "Office Hours" module.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * @addtogroup hooks
+ * @{
+ */
+
+/**
+ * Allows to alter the current time of isOpen() method.
+ *
+ * @param int $time
+ * @param \Drupal\Core\Entity\EntityInterface $node
+ */
+function hook_office_hours_isopen_time_alter(int &$time, EntityInterface $entity) {
+  // Assume that all owned entities (for example, offices and ad) should
+  // depends on user timezone. Add user timezone to time.
+  if ($entity instanceof \Drupal\user\EntityOwnerInterface
+    && ($account = $entity->getOwner())
+    && ($account->isAuthenticated())
+  ) {
+    $date_time = new \DateTime();
+    $date_time->setTimestamp($time);
+    $date_time->setTimezone(new \DateTimeZone($account->getTimeZone()));
+    $time = $date_time->getTimestamp();
+  }
+}
\ No newline at end of file
diff --git a/src/Plugin/Field/FieldType/OfficeHoursItemList.php b/src/Plugin/Field/FieldType/OfficeHoursItemList.php
index cc6a9a6..0eb7e65 100644
--- a/src/Plugin/Field/FieldType/OfficeHoursItemList.php
+++ b/src/Plugin/Field/FieldType/OfficeHoursItemList.php
@@ -19,6 +19,11 @@ class OfficeHoursItemList extends FieldItemList implements OfficeHoursItemListIn
     // Detect the current line and the open/closed status.
     // Convert the day_number to (int) to get '0' for Sundays, not 'false'.
     $time = ($time === NULL) ? \Drupal::time()->getRequestTime() : $time;
+
+    // Allows to alter the current time.
+    $entity = $this->getEntity();
+    \Drupal::moduleHandler()->alter('office_hours_isopen_time', $time, $entity);
+
     $today = (int) idate('w', $time); // Get day_number sun=0 - sat=6.
     $now = date('Hi', $time); // 'Hi' format, with leading zero (0900).
     $is_open = FALSE;
